app/gui/src/dashboard/hooks/backendHooks.ts

```diff
@@ -651,6 +651,25 @@ export function useNewSecret(backend: Backend) {
   })
 }
 
+/** A function to create a new credential. */
+export function useNewCredential(backend: Backend) {
```

Comment:
I don't think we need a hook here, prefer `newCredentialMutationOptions` and use an appropriate hook in place `useMutation/useMutationCallback`